### ISSUE TITLE
Add HSTS and HPKP

### ISSUE THREAD

**Issue by blobgo** [sibiantony/ssleuth#24]
It would be good to also check whether the server is using [HTTP Strict Transport Security] (https://en.wikipedia.org/wiki/HTTP_Strict_Transport_Security) and [HTTP Public Key Pinning] (HTTP Public Key Pinning).

**Comment by tr37ion** [comment]
This would be a fine addition.

**Comment by sibiantony** [comment]
Indeed, this is in the roadmap. Please expect some delay, too busy right now.

**Comment by tr37ion** [comment]
Great, thanks for your work.

**Comment by ekaris** [comment]
When will this be implemented? HSTS shouldn't be too difficult to implement

**Comment by sibiantony** [comment]
@ekaris at the moment I'm looking to e10s support in my spare time. I hope to complete it before mid 2016. As soon as that is done, I can incorporate other features like this.
But PRs are always welcome!

**Comment by ekaris** [comment]
I can try to see if I can do this, it doesn't look much effort, just check if header is present

**Comment by shirishag75** [comment]
@ekaris just saw you put a PR at https-everywhere, so would urge you to fork and make a PR for this as well. Maybe something that could be built as an experimental mode - wdyt @sibiantony

**Comment by ekaris** [comment]
I'm going to leave this to the dev since the addon would need to be rewritten using WebExtensions next year to continue to work.

**Comment by rugk** [comment]
We'll see. I'd say Mozilla can't stick to the schedule, when we see how much the e10s project has already been delayed, because of incompatible add-ons.
